Sources/IdentityState.swift

```diff
@@ -94,6 +94,32 @@ class IdentityState {
         createXDMSharedState(identityProperties.toXdmData(), nil)
         return hasBooted
     }
+    
+    /// When the advertising identifier from the `event` is different from the current value, it updates the persisted value and creates
+    /// new shared state and XDM shared state. A consent request event is dispatched when advertising tracking preferences change.
+    /// If privacy is optedout the call is ignored
+    /// - Parameters:
+    ///   - event: event containing a new ADID value.
+    ///   - createXDMSharedState: function which creates new XDM shared state
+    ///   - eventDispatcher: function which dispatchs events to the event hub
+    func updateAdvertisingIdentifier(event: Event,
+                                     createXDMSharedState: ([String: Any], Event) -> Void,
+                                     eventDispatcher: (Event) -> Void) {
+
+        // update adid if changed and extract the new adid value
+        let (adIdChanged, shouldUpdateConsent) = shouldUpdateAdId(newAdID: event.adId)
+        if adIdChanged, let adId = event.adId {
```

Comment:
For the second condition, we should allow nil values for event.adId and update the identityProperties value below